lib/matplotlib/tests/test_backend_qt.py

```diff
@@ -272,3 +272,19 @@ def test_figureoptions():
             "matplotlib.backends.qt_editor._formlayout.FormDialog.exec_",
             lambda self: None):
         fig.canvas.manager.toolbar.edit_parameters()
+
+
+@pytest.mark.backend('Qt5Agg')
+def test_double_resize():
+    # Check that resizing a figure twice keeps the same window size
+    fig, ax = plt.subplots()
+    fig.canvas.draw()
+    window = fig.canvas.manager.window
+
+    fig.set_size_inches(2, 2)
+    old_width = window.width()
```

Comment:
In my test of this code, the example window is always 640x480 no matter what I pass to `set_size_inches`. Maybe explicitly test for 200 here (or `rcParams['figure.dpi'] * 2` if you want to be very explicit).